testinfra/modules/docker.py

```diff
@@ -0,0 +1,47 @@
+# coding: utf-8
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from testinfra.modules.base import Module
+
+
+class Docker(Module):
+    """Test docker containers running on system.
+
+    Example:
+
+    >>> host = testinfra.get_host('local://')
+
+    >>> nginx = host.docker("nginx:latest")
+    >>> assert nginx.is_running
+    """
+
+    def __init__(self, name):
+        self.name = name
+        super(Docker, self).__init__()
+
+    @property
+    def is_running(self):
+        cmd = self.run("docker ps -q -f ancestor=%s --format '{{.Image}}'",
+                       self.name)
```

Comment:
Sorry missed that on first review. I think testing "image" is running is wrong because you can have multiple running containers with the same image, also if a new ``nginx:latest`` is pulled I'm not sure ``{{.Image}}`` will remain the same.

I'd rather parse the output of ``docker ps --format '{{ .ID }} {{ .Names }}'`` and find ``self.name`` in the list of docker ids and names.

Reply:
Hey @philpep I wasn't sure what to exactly check for. It could be debatable that sometimes you might want to know if an instance of an image is running and want to assert on that, especially if you know the name of the image doesn't change. If you are asserting on the name of the running container, i.e. sleepy_hare, you would have to change your code each time you spawn a different one.

It would make sense though to assert on the running container name though, if that container is going to exist for a while. I changed the code to reflect that, based on your suggestions. Now you should be able to perform the actions you specified, additionally you can get the image id by using

```nginx.image_id```

I think for future we can implement another class called "docker_image" that will represent the image info, since "docker inspect image_id" returns some useful stuff as well. With those two classes you should be able to do so something like

```assert len(host.docker_image("nginx:latest").running_instances) > 0```

Reply:
What about adding a classmethod to return the list of `Docker` instances running ? Eventually with a parameter to filter containers.

```
>>> host.docker.get_containers()
[Docker('XXX'), Docker('YYY')]
>>> host.docker;get_containers(image="nginx:latest")
[Docker("XXX")]
```

I think it's better to do this in a following PR, thanks for updating this PR, I think I can merge part of it, that will give us a good base to work on.